packages/@aws-containers/aws-ecs-builder/lib/environment.ts

```diff
@@ -0,0 +1,82 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as cdk from '@aws-cdk/core';
+import { EnvironmentCapacityType } from './extensions/extension-interfaces';
+
+/**
+ * Settings for the deploy environment you want to deploy
+ * services within.
+ */
+export interface EnvironmentProps {
+  /**
+   * The VPC used by the service for networking
+   * @default - Create a new VPC
+   */
+  readonly vpc?: ec2.IVpc,
+
+  /**
+   * The ECS cluster which provides compute capacity to this service
+   * [disable-awslint:ref-via-interface]
+   * @default - Create a new cluster
+   */
+  readonly cluster?: ecs.Cluster
+
+  /**
+   * What type of capacity to use for this environment
+   * @default - EnvironmentCapacityType.FARGATE
+   */
+  readonly capacityType?: EnvironmentCapacityType
+}
+
+/**
+ * A service builder class. This construct support various extensions
+ * which can construct an ECS service progressively.
+ */
+export class Environment extends cdk.Construct {
```

Comment:
(not specifically for this PR) Have you given any thought to what it might look like connecting two Environments together? Seems like another area where we could expose a few new patterns, such as a VPC endpoint and App Mesh gateway.

Reply:
Good call out. I discovered an edgecase where people could still attempt to connect services from two different environments to each other even though that isn't supported yet. I added a error message when this is attempted. In a future release we will want to design an implementation for making this a possibility (VPC peering, App Mesh peering, etc). For now the initial release will assume each environment is fully standalone from other environments.